nyx_client/nyx_client/circles.py

```diff
@@ -0,0 +1,113 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class RemoteHost:
+    """Represents a remote host on the network."""
+
+    def __init__(self, name: str, did: str):
+        """Builds a RemoteHost.
+
+        Args:
+            name: the name of the organization
+            did: the did of the remote host
+        """
+        self.name = name
+        self.did = did
+
+    def __repr__(self):
+        return f"RemoteHost({self.name}, {self.did})"
+
+
+class Circle:
+    """Represents a circle, which is a grouping of remote hosts."""
+
+    def __init__(
+        self, did: str, name: str, description: str | None = None, organizations: list[RemoteHost] | None = None
+    ):
+        """Builds a circle.
+
+        Args:
+            did: the did of the circle
+            name: the name of the circle
+            description: optional description of what the circle is
+            organizations: optional list of remote organizations in the circle
+        """
+        self.did = did
+        self.name = name
+        self.description = description
+        self.organizations = organizations
+
+    def __repr__(self):
+        return f"Circle({self.name}, {self.description}, {self.organizations})"
+
+    def contains(self, to_check: RemoteHost | str) -> bool:
+        """Checks if provided host is in the circle.
+
+        Args:
+            to_check: Either the RemoteHost to check, or a DID, or org name.
+
+        Returns:
+            Boolean showing if the remote host is included.
+        """
+        if not self.organizations:
+            return False
+        if isinstance(to_check, str):
+            for host in self.organizations:
+                if to_check == host.did or to_check == host.name:
+                    return True
+        elif isinstance(to_check, RemoteHost):
+            return to_check in self.organizations
+
+        return False
+
+    @classmethod
+    def from_json(cls, value: dict):
+        """Builds a circle object from json.
+
+        Args:
+            value: the json dictionary (returned from API).
+
+        Returns:
+            Circle object
+        """
+        allowed_hosts: list[RemoteHost] = []
+        for raw_host in value.get("organizations", []):
+            allowed_hosts.append(RemoteHost(did=raw_host["did"], name=raw_host["name"]))
+        return cls(
+            did=value["did"],
+            name=value["name"],
+            description=value.get("description"),
+            organizations=allowed_hosts,
+        )
+
+    def as_dict(self) -> dict:
+        """Returns the object as a dictionary.
+
+        Returns:
+            A dictionary of the circle, that matches POST/PUT requests in the API.
+
+        """
+        resp: dict[str, str | list] = {
```

Comment:
(unimportant) - I think what you're looking here for is a typed dict.
But: Could just instead use `dataclasses.asdict` (and wouldn't need this at all?)